main.go

```diff
@@ -73,10 +74,11 @@ func (s *Searcher) Load(filename string) error {
 }
 
 func (s *Searcher) Search(query string) []string {
-	idxs := s.SuffixArray.Lookup([]byte(query), -1)
+	r, _ := regexp.Compile("(?i)" + query)
+	idxs := s.SuffixArray.FindAllIndex(r, 20)
```

Comment:
To fix the 2nd test I decided to use regex. I changed the method on the SuffixArray to utilize the regex created on line 77 by combining a case insensitive character and the query string. 

This `FindAllIndex` method returns pairs: [1st char of query string, last char of query string] so on line 81, I used the 1st item in each tuple as the center point for each search result. 

I also made the change here for test 3. Changing the 2nd argument of the SuffixArray method to return the first 20 results.